components/event-bus/internal/knutil/eventing.go

```diff
@@ -0,0 +1,270 @@
+package knutil
+
+import (
+	"bytes"
+	"errors"
+	"crypto/tls"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	evclientset "github.com/knative/eventing/pkg/client/clientset/versioned"
+	evapisv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	eventingv1alpha1 "github.com/knative/eventing/pkg/client/clientset/versioned/typed/eventing/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/rest"
+
+)
+
+/*
+//
+// sample usage of knutil
+//
+
+// get a channel
+var ch *eventingv1alpha1.Channel
+if ch, err = GetChannel(channelName, namespace); err != nil && k8serrors.IsNotFound(err) {
+	// channel doesn't exist, create it
+	if ch, err = CreateChannel(provisionerName, channelName, namespace); err != nil {
+		log.Printf("ERROR: createChannel() failed: %v", err)
+		return
+	}
+} else if err != nil {
+	log.Printf("ERROR: getChannel() failed: %v", err)
+	return
+}
+
+// send a message to channel
+var msg = "test-message"
+if err := SendMessage(ch, "&msg); err != nil {
+	log.Printf("ERROR: sendMessage() failed: %v", err)
+	return
+}
+
+// create a subscription
+var uri = "dnsName: hello-00001-service.default"
+if err := CreateSubscription("my-sub", namespace, channelName, &uri); err != nil {
+	log.Printf("ERROR: create subscription failed: %v", err)
+	return
+}
+return
+*/
+
+
+// GetChannel returns an existing Knative/Eventing channel, if it exists.
+// If the channel doesn't exist, the error returned can be checked using the
+// standard K8S function: "k8serrors.IsNotFound(err) "
+// Sample usage:
+// if ch, err = GetChannel(channelName, namespace); err != nil && k8serrors.IsNotFound(err) {
+//    // channel doesn't exists, must be created first
+// } else if err != nil {
+//    // other errors
+// }
+func GetChannel(name string, namespace string) (*evapisv1alpha1.Channel, error) {
+	config, err := rest.InClusterConfig()
```

Comment:
Should we consider doing this only once like the initial setup as we do for push?

Reply:
yes

Reply:
I have the same recommendation as Gaurav.